pkg/distributor/distributor.go

```diff
@@ -504,10 +496,28 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 		ingestionRateStrategy = newGlobalRateStrategyWithBurstFactor(limits, d)
 	}
 
+	// If this isn't a real distributor that will be accepting writes or if the HA tracker is
+	// disabled, use a no-op implementation. We don't want to require it to be configured when
+	// it's not used due to being disabled or this being a component that doesn't use it (rulers,
+	// queriers).
+	var haTrackerImpl haTracker
+
+	if !canJoinDistributorsRing || !cfg.HATrackerConfig.EnableHATracker {
+		haTrackerImpl = newNopHaTracker()
```

Comment:
I was thinking also if we should wrap the behaviour in the above if statement, to avoid duplicate check. wdyt ? 

```
if !canJoinDistributorsRing {
 requestRateStrategy = newInfiniteRateStrategy()
....
```

Reply:
I have this as its own block on purpose since the logic is slight different and the justification (comment) is different than the other block.